### DISCUSSION TITLE
Set remote temperature

### DISCUSSION THREAD

**Discussion by johnhoran** [gekkekoe/esphome-ecodan-hp#49] [General]
Just a question, but I see a number of projects using the swicago library to set current room temperature in the heatpump.

https://github.com/SwiCago/HeatPump/blob/5d1e146771d2f458907a855bf9d5d4b9bf5ff033/src/HeatPump.cpp#L305C1-L330C2
https://github.com/tobias-93/esphome-ecodan-heatpump/blob/2fdf15ba675699c0c2ca8c1692a31505cb02cc8f/README.md?plain=1#L106-L153
(though they do say they weren't able to set it in the readme)

I'm wondering if you ever looked at this and if it were supported, would the heatpump modulate based off this?

**Comment by gekkekoe** [comment]
Please ask questions in discussion section (moved this question)

I did actually try it, but it did not work, seems to work for airco's. The remote thermostat is connected on a different port, but they do speak a similar protocol.

**Comment by TimoFriedri** [comment]
Has anyone looked into the idea of using the second UART to speak to the CNRF port of air-to-water Mitsubishi heatpumps.

I read somewhere that the electronics on a FTC6 for this port are identical to CN105, and it is assumed it talks the same protocol.


FTC6 controller boards are known to have a terrible simple Heat Curve Control mode. Only the Room Target Temperature Mode support proper hysteresis etc. It would be very beneficial to be able to set the remote temperature via ESP and Home Assistant.

**Reply by DamijanD** [reply]
> FTC6 controller boards are known to have a terrible simple Heat Curve Control mode. Only the Room Target Temperature Mode support proper hysteresis etc. It would be very beneficial to be able to set the remote temperature via ESP and Home Assistant.

@TimoFriedri if I understand correctly you would like to send to FTC current room temperature (so instead of optional room sensor). Like it was explained earlier that is not possible via CN105. (I have 2 zones with 2 remote controls/sensors, but only one zone can use adjusted mode. The other must be in a simple weather curve mode. And for this zone room temperature is used only in some rare cases - when to start heating if the zone is in idle state).
But what you can probably do: use the set flow temperature mode and then you can probably set desired flow temperature via CN105, based on other data that is available to you: outside temp, room temp, flow and return flow temp. (I think that you can set this temperature via melcloud, but I am not 100% sure)

**Comment by gekkekoe** [comment]
We already use another uart for the proxy, the esp32 should have 3 in total.

Issue seems that the protocol for set remote temp is a bit different. I'm not saying it's impossible, but is some work. Also, I don't have a remote thermostat, so I cannot test.

In other words, it does not have a lot of priority on my todo list, but if someone is up for the task, try and create a PR when it works and I will review it.

**Comment by F1p** [comment]
CNRF does not send or respond to the same connect commands as CN105 on my FTC5

**Comment by gekkekoe** [comment]
I read somewhere that rx/tx was swapped on the CNRF port. Some one could try and run proxy mode on the CNRF port and trace how it connects

**Comment by TimoFriedri** [comment]
Just for reference. Here people discuss CNRF a little: https://gathering.tweakers.net/forum/list_messages/2112892/115

@gekkekoe  I assume you already know this forum seeing your repo linked in other threads.

**Comment by gekkekoe** [comment]
Some update on this, me and @F1p can act like a remote thermostat now, stay tuned :)
We should be able to support 8 remote thermostats and feed them with arbitrary temps

**Reply by DamijanD** [reply]
How HP uses more than one temp sensor per zone?

**Reply by gekkekoe** [reply]
you can configure up to 8 remote thermostat for different rooms. But only one has to be picked as main.

**Comment by gekkekoe** [comment]
@johnhoran @TimoFriedri 
Project is available, use this for remote thermostat functionality:
https://github.com/gekkekoe/esphome-ecodan-remote-thermostat

**Reply by ruimarinho** [reply]
@gekkekoe I am looking to get into the Ecodan HP as well for radiant floor heating + DWH. I am planning on deploying my own thermostats via RS485 which, from my understanding, I can feed back into the Ecodan with that project. How exactly does that improve the Ecodan HP management?

Thank you for all the work in these projects. It's the only reason why I'm not even considering Daikin at this point.

**Reply by gekkekoe** [reply]
The remote thermostats (PARxx or my project) and ftc controller are the only units that can be used for auto adaptive temp feature. 

With my project you can use any sensor (that can be interfaced in HA) as remote thermostat. You can control how you would like the hysteresis to work. Basically more freedom

**Reply by DamijanD** [reply]
Beside that you can also put main controller (the one that is inside the inner unit) into the living room and it can server as a termostat as well.

**Reply by gekkekoe** [reply]
yeah thats what I meant with "FTC controller"

**Reply by DamijanD** [reply]
my bad :-)
